packages/block-library/src/cover-image/index.js

```diff
@@ -117,146 +127,171 @@ export const settings = {
+			const setDimRatio = ( ratio ) => setAttributes( { dimRatio: ratio } );
 
-		const style = backgroundImageStyles( url );
-		const classes = classnames(
-			className,
-			contentAlign !== 'center' && `has-${ contentAlign }-content`,
-			dimRatioToClass( dimRatio ),
-			{
-				'has-background-dim': dimRatio !== 0,
-				'has-parallax': hasParallax,
-			}
-		);
+			const style = {
+				...backgroundImageStyles( url ),
+				backgroundColor: overlayColor.color,
+			};
 
-		const controls = (
-			<Fragment>
-				<BlockControls>
-					<BlockAlignmentToolbar
-						value={ align }
-						onChange={ updateAlignment }
-					/>
-					<AlignmentToolbar
-						value={ contentAlign }
-						onChange={ ( nextAlign ) => {
-							setAttributes( { contentAlign: nextAlign } );
-						} }
+			const classes = classnames(
+				className,
+				contentAlign !== 'center' && `has-${ contentAlign }-content`,
+				dimRatioToClass( dimRatio ),
+				{
+					'has-background-dim': dimRatio !== 0,
+					'has-parallax': hasParallax,
+				}
+			);
+
+			const controls = (
+				<Fragment>
+					<BlockControls>
+						<BlockAlignmentToolbar
+							value={ align }
+							onChange={ updateAlignment }
+						/>
+						<AlignmentToolbar
+							value={ contentAlign }
+							onChange={ ( nextAlign ) => {
+								setAttributes( { contentAlign: nextAlign } );
+							} }
+						/>
+						<Toolbar>
+							<MediaUpload
+								onSelect={ onSelectImage }
+								allowedTypes={ ALLOWED_MEDIA_TYPES }
+								value={ id }
+								render={ ( { open } ) => (
+									<IconButton
+										className="components-toolbar__control"
+										label={ __( 'Edit image' ) }
+										icon="edit"
+										onClick={ open }
+									/>
+								) }
+							/>
+						</Toolbar>
+					</BlockControls>
+					{ !! url && (
+						<InspectorControls>
+							<PanelBody title={ __( 'Cover Image Settings' ) }>
+								<ToggleControl
+									label={ __( 'Fixed Background' ) }
+									checked={ !! hasParallax }
+									onChange={ toggleParallax }
+								/>
+								<PanelColorSettings
+									title={ __( 'Overlay' ) }
+									initialOpen={ true }
+									colorSettings={ [ {
+										value: overlayColor.color,
+										onChange: setOverlayColor,
+										label: __( 'Overlay Color' ),
+									} ] }
+								>
+									<RangeControl
+										label={ __( 'Background Opacity' ) }
+										value={ dimRatio }
+										onChange={ setDimRatio }
+										min={ 0 }
+										max={ 100 }
+										step={ 10 }
+									/>
+								</PanelColorSettings>
+							</PanelBody>
+						</InspectorControls>
+					) }
+				</Fragment>
+			);
+
+			if ( ! url ) {
+				const hasTitle = ! RichText.isEmpty( title );
+				const icon = hasTitle ? undefined : 'format-image';
+				const label = hasTitle ? (
+					<RichText
+						tagName="h2"
+						value={ title }
+						onChange={ ( value ) => setAttributes( { title: value } ) }
```

Comment:
I dunno if we have a policy on inline functions as props but my brain always thinks "technically we're creating a new function on every render". It's also less easy to test/identify in stack traces/etc. when it's an inline anonymous function.

All of that doesn't really matter here, but it's just a thought. I like being able to see what functions exist in a component without having to go way down into props.

`</NitpickOfTheCentury>`

🤷‍♂️

Reply:
Related discussion on inline functions, at least when it comes to performance: 
https://github.com/WordPress/gutenberg/pull/5495#discussion_r192355655